sycl/doc/DynamicLinkingTestPlan.md

```diff
@@ -0,0 +1,174 @@
+# Test plan for dynamic linking of device code
+
+This is a test plan for feature dynamic linking of device code. Feature
+description is available in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/SharedLibraries.md .
+
+## Testing scope
+
+### Device & BE coverage
+
+All of the tests described below are performed on CPU and GPU devices.
+For GPU device both OpenCL and L0 backends are used.
+
+**NOTE**: The feature won't be supported on FPGA device.
+
+### AOT & JIT compilation
+
+All the tests are compiled using AOT, JIT and mixed (AOT with JIT) compilation.
+
+### Device code split modes
+
+All the tests are compiled using different combination of device code split
+mode.
+There are four possible device code split modes that can be requested via
+`fsycl-device-code-split` compiler's option:
+- `per_kernel` - device code module is created for each SYCL kernel
+- `per_source` - device code module is created for each source
+- `off` - no device code split
+- `auto` - use heuristic to select the best way of splitting device code
+
+For each test the main application and all shared libraries are compiled with
+all the variations of device code split modes. This results in four binaries for
+each application or library.
+Each tests covers all the combinations of device code split modes used for
+the application and libraries, i.e. all binaries of the application are checked
+with all binaries for shared libraries.
+For one application and one shared library examples of the combinations will be:
+- Application compiled with `per_kernel` and library compiled with `per_kernel`
+- Application compiled with `per_kernel` and library compiled with `per_source`
+- Application compiled with `per_kernel` and library compiled with `off`
+- Application compiled with `per_kernel` and library compiled with `auto`
+
+- Application compiled with `per_source` and library compiled with `per_kernel`
+- Application compiled with `per_source` and library compiled with `per_source`
+- Application compiled with `per_source` and library compiled with `off`
+- Application compiled with `per_source` and library compiled with `auto`
+
+and etc. Total 16 combinations.
+
+## Tests
+
+All the tests consist of the main application and at least one shared library.
+For all the tests simple calculation is performed by the application's kernel(s)
+that used `SYCL_EXTERNAL` from the shared library. Example : addition of
+vectors.
+The result of the calculation must be checked on host.
+
+### Basic test with shared library
+
+- Several kernels and `SYCL_EXTERNAL` functions are defined in both library and
+  application.
+- At least one `SYCL_EXTERNAL` function defined in shared library is used from
+  the application.
+- At least one `SYCL_EXTERNAL` function defined in shared library is used in
+  kernel inside the library.
+- Defined kernels are submitted and ran.
```

Comment:
It can, if they are used from `SYCL_EXTERNAL` or a kernel inside the library. However In current design they are not considered as imported/exported symbols and therefore don't participate to symbols resolution process.